From 364146de8b6c39eeeb30361338aa1ce5cab78e1a Mon Sep 17 00:00:00 2001
From: Dmitri Smirnov <d.s@plexoos.com>
Date: Mon, 21 Apr 2014 19:20:29 -0400
Subject: [PATCH] New fake detector id for test geometries

---
 StEvent/StEnumerations.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/StEvent/StEnumerations.h b/StEvent/StEnumerations.h
index eb62055..ce19bb7 100644
--- a/StEvent/StEnumerations.h
+++ b/StEvent/StEnumerations.h
@@ -241,6 +241,7 @@ enum StDetectorId {kUnknownId   = kUnknownIdentifier,
                    kPhmdId      = kPhmdIdentifier,
                    kPxlId       = kPxlIdentifier,
                    kIstId       = kIstIdentifier,
+                   kTestGeomId,
                    kFgtId       = kFgtIdentifier,
                    kEtrId       = kEtrIdentifier,                         // 32
                    kFpdWestId   = kFpdWestIdentifier,
-- 
1.8.2.GIT

