From 29dfcc6c4dcfbf8565cabb91250e56bf0f0adf45 Mon Sep 17 00:00:00 2001
From: Dmitri Smirnov <d.s@plexoos.com>
Date: Tue, 29 Apr 2014 14:24:36 -0400
Subject: [PATCH 1/6] Individual options for making HFT hits

---
 BigFullChain.h | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/BigFullChain.h b/BigFullChain.h
index a011622..fcf2869 100644
--- a/BigFullChain.h
+++ b/BigFullChain.h
@@ -1246,7 +1246,6 @@ Bfc_st BFC[] = {
   {"eemcDb"      ,"eeDb" ,"","db,EEmcUtil",      "StEEmcDbMaker","StEEmcDbMaker","Load EEmcDbMaker",kFALSE},
   {"fmsDb"       ,"fmsDb","","db",                  "StFmsDbMaker","StFmsDbMaker","Load FmsDbMaker",kFALSE},
   {"fgtDb"       ,"fgtDb","","db,fgtutil",          "StFgtDbMaker","StFgtDbMaker","Load FgtDbMaker",kFALSE},
-  {"pxlDb"       ,"pxlDb","","tpcDb",               "StPxlDbMaker","StPxlDbMaker","Load PxlDbMaker",kFALSE},
   {"istDb"       ,"istDb","","tpcDb",               "StIstDbMaker","StIstDbMaker","Load IstDbMaker",kFALSE},
 
   {"------------","-----------","-----------","------------------------------------------","","","",kFALSE},
@@ -1361,6 +1360,18 @@ Bfc_st BFC[] = {
   {"Event"       ,  "","","MakeEvent",                          "","","Request to initialize event",kFALSE},
   {"pxlFastSim","","","StMcEvent,StEvent"
    ,                               "StPxlSimMaker","StPxlSimMaker","FastPixelSimulator",kFALSE},
+  {"pxlDb"      , "" , "" , "pxlUtil tpcDb"       , "StPxlDbMaker"          , "StPxlDbMaker"                            , "Load and run StPxlDbMaker"          , kFALSE},
+  {"pxlRaw"     , "" , "" , "pxlUtil"             , "StPxlRawHitMaker"      , "StPxlRawHitMaker"                        , "Load and run StPxlRawHitMaker"      , kFALSE},
+  {"pxlCluster" , "" , "" , "pxlUtil"             , "StPxlClusterMaker"     , "StPxlClusterMaker"                       , "Load and run StPxlClusterMaker"     , kFALSE},
+  {"pxlHit"     , "" , "" , "pxlUtil event pxlDb" , "StPxlHitMaker"         , "StPxlHitMaker"                           , "Load and run StPxlHitMaker"         , kFALSE},
+  {"pxlMon"     , "" , "" , "StEvent"             , "StPxlMonMaker"         , "StPxlMonMaker"                           , "Produce QA histograms for PXL"      , kFALSE},
+  {"istFastSim" , "" , "" , "StMcEvent StEvent"   , "StIstFastSimMaker"     , "StIstFastSimMaker"                       , "Load and run StFastIstSimulator"    , kFALSE},
+  {"istRaw"     , "" , "" , "istUtil istDb"       , "StIstRawHitMaker"      , "StIstRawHitMaker"                        , "Load and run StIstRawHitMaker"      , kFALSE},
+  {"istCluster" , "" , "" , "istUtil"             , "StIstClusterMaker"     , "StIstClusterMaker"                       , "Load and run StIstClusterMaker"     , kFALSE},
+  {"istHit"     , "" , "" , "istUtil event istDb" , "StIstHitMaker"         , "StIstHitMaker"                           , "Load and run StIstHitMaker"         , kFALSE},
+  {"istQA"      , "" , "" , "istUtil StEvent"     , "StIstQAMaker"          , "StIstQAMaker"                            , "Produce QA histograms for IST"      , kFALSE},
+  {"istCalib"   , "" , "" , "istUtil"             , "StIstCalibrationMaker" , "StIstCalibrationMaker"                   , "Load and run StIstCalibrationMaker" , kFALSE},
+  {"HftMatTree" , "" , "" , "pxlUtil"             , "HftMatchedTree"        , "StHftPoolEventT StHftPoolHftMatchedTree" , "Create HftMatchedTree"              , kFALSE},
   {"ssddat"      ,"","","ssd_daq"                             ,"","","SSD full chain for Real Data",kFALSE},
   {"ssd_daq","","","ssdCalDb,svt_T,-sls,-spa,ssdUtil","StSsdDaqMaker","StSsdDaqMaker","... SSD Daq",kFALSE},
   {"ssdfast"     ,"","","ssdDb,StMcEvent,StEvent","StSsdFastSimMaker","StSsdFastSimMaker",
-- 
1.8.2.GIT


From 76fd05690c240ccb1ad8fa43a48a47e645249057 Mon Sep 17 00:00:00 2001
From: Dmitri Smirnov <d.s@plexoos.com>
Date: Mon, 21 Apr 2014 19:01:43 -0400
Subject: [PATCH 2/6] BFC options for building test geometries

---
 BigFullChain.h |  8 ++++++++
 StBFChain.cxx  | 13 +++++++++++++
 2 files changed, 21 insertions(+)

diff --git a/BigFullChain.h b/BigFullChain.h
index fcf2869..01b5355 100644
--- a/BigFullChain.h
+++ b/BigFullChain.h
@@ -1472,6 +1472,13 @@ Bfc_st BFC[] = {
   {"StiSst"   ,"","",""                        ,"","StSsdUtil,StSsdDbMaker,StiSsd","Load StiSsd(t)",kFALSE},
   {"StiPxl"   ,"","",""                           ,"","StPxlUtil,StPxlDbMaker,StiPxl","Load StiPxl",kFALSE},
   {"StiIst"   ,"","",""                           ,"","StIstUtil,StIstDbMaker,StiIst","Load StiIst",kFALSE},
+  {"StiTestSimpleBox", "", "", "", "",              "", "Flag to build specific test Sti geometry", kFALSE},
+  {"StiTestSimpleBox2", "", "", "", "",             "", "Flag to build specific test Sti geometry", kFALSE},
+  {"StiTestSimpleBox8", "", "", "", "",             "", "Flag to build specific test Sti geometry", kFALSE},
+  {"StiTestSimplePlane", "", "", "", "",            "", "Flag to build specific test Sti geometry", kFALSE},
+  {"StiTestSimpleTube", "", "", "", "",             "", "Flag to build specific test Sti geometry", kFALSE},
+  {"StiTestSimpleTubeSector", "", "", "", "",       "", "Flag to build specific test Sti geometry", kFALSE},
+  {"StiTestEnclosedTubes", "", "", "", "",          "", "Flag to build specific test Sti geometry", kFALSE},
 
   // left for historical purpose
   {"StiLibs"    ,"","","StiTpc,StiSvt,StiSsd,BTofUtil"   
@@ -1514,6 +1521,7 @@ Bfc_st BFC[] = {
   {"PxlIT"  ,""  ,"","ITTF",""                             ,"Sti,StiPxl","Sti tracking: Pixel geom",kFALSE},
   {"IstIT"  ,""  ,"","ITTF",""                               ,"Sti,StiIst","Sti tracking: Ist geom",kFALSE},
   {"SstIT"  ,""  ,"","ITTF",""                               ,"Sti,StiSst","Sti tracking: Sst geom",kFALSE},
+  {"TestIT" ,""  ,"","ITTF",""                             ,"Sti,StiTest","Sti tracking: Test geom",kFALSE},
 
   {"BTofIT"  ,""  ,"","ITTF",""                            ,"Sti,StiBTof","Sti tracking: BTof geom",kFALSE},
   {"NoSvtIT"     ,""  ,"","-SvtIT",""                    ,"","ITTF: track with switch off SVT geom",kFALSE},
diff --git a/StBFChain.cxx b/StBFChain.cxx
index 62d6978..37d2b08 100644
--- a/StBFChain.cxx
+++ b/StBFChain.cxx
@@ -462,6 +462,19 @@ Int_t StBFChain::Instantiate()
 	mk->SetAttr("useSst"	 ,kTRUE);
 	mk->SetAttr("activeSst"  ,kTRUE);
       }
+      if (GetOption("TestIT")){
+
+         if      (GetOption("StiTestSimpleBox"))        mk->SetAttr("StiTestSimpleBox", true);
+         else if (GetOption("StiTestSimpleBox2"))       mk->SetAttr("StiTestSimpleBox2", true);
+         else if (GetOption("StiTestSimpleBox8"))       mk->SetAttr("StiTestSimpleBox8", true);
+         else if (GetOption("StiTestSimplePlane"))      mk->SetAttr("StiTestSimplePlane", true);
+         else if (GetOption("StiTestSimpleTube"))       mk->SetAttr("StiTestSimpleTube", true);
+         else if (GetOption("StiTestSimpleTubeSector")) mk->SetAttr("StiTestSimpleTubeSector", true);
+         else if (GetOption("StiTestEnclosedTubes"))    mk->SetAttr("StiTestEnclosedTubes", true);
+
+         mk->SetAttr("useTestGeom", kTRUE);
+         mk->SetAttr("activeTestGeom", kFALSE);
+      }
 
       // other sub-systems
       if (GetOption("BTofIT")){
-- 
1.8.2.GIT


From 290c9de0ad6f90a80f3e5e52276db5e1e644332c Mon Sep 17 00:00:00 2001
From: Dmitri Smirnov <d.s@plexoos.com>
Date: Tue, 22 Apr 2014 16:27:17 -0400
Subject: [PATCH 3/6] Change to get rid of doubly defined StiMaker attributes

---
 StBFChain.cxx | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/StBFChain.cxx b/StBFChain.cxx
index 37d2b08..9aadc68 100644
--- a/StBFChain.cxx
+++ b/StBFChain.cxx
@@ -425,6 +425,7 @@ Int_t StBFChain::Instantiate()
 
       // old logic for svt and ssd
       if (GetOption("NoSvtIT")){
+	mk->SetAttr("useSvt"	, ".remove");
 	mk->SetAttr("useSvt"	,kFALSE);
       } else {
 	if (GetOption("SvtIT")){
@@ -434,6 +435,7 @@ Int_t StBFChain::Instantiate()
       }
       if (   GetOption("NoSsdIT") &&
 	    !GetOption("SstIT") ){
+	mk->SetAttr("useSsd"	,".remove");
 	mk->SetAttr("useSsd"	,kFALSE);
       } else {
 	if (GetOption("SsdIT")){
-- 
1.8.2.GIT


From 08b25dff353f59eb03e54e058366613e3f3667aa Mon Sep 17 00:00:00 2001
From: Dmitri Smirnov <d.s@plexoos.com>
Date: Thu, 24 Apr 2014 16:25:31 -0400
Subject: [PATCH 4/6] Added new options for Sti test to check overlapping and
 non-overlapping tubes

---
 BigFullChain.h | 2 ++
 StBFChain.cxx  | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/BigFullChain.h b/BigFullChain.h
index 01b5355..91921fc 100644
--- a/BigFullChain.h
+++ b/BigFullChain.h
@@ -1479,6 +1479,8 @@ Bfc_st BFC[] = {
   {"StiTestSimpleTube", "", "", "", "",             "", "Flag to build specific test Sti geometry", kFALSE},
   {"StiTestSimpleTubeSector", "", "", "", "",       "", "Flag to build specific test Sti geometry", kFALSE},
   {"StiTestEnclosedTubes", "", "", "", "",          "", "Flag to build specific test Sti geometry", kFALSE},
+  {"StiTestOverlapTubes", "", "", "", "",           "", "Flag to build specific test Sti geometry", kFALSE},
+  {"StiTestNoOverlapTubes", "", "", "", "",         "", "Flag to build specific test Sti geometry", kFALSE},
 
   // left for historical purpose
   {"StiLibs"    ,"","","StiTpc,StiSvt,StiSsd,BTofUtil"   
diff --git a/StBFChain.cxx b/StBFChain.cxx
index 9aadc68..fd39171 100644
--- a/StBFChain.cxx
+++ b/StBFChain.cxx
@@ -473,6 +473,8 @@ Int_t StBFChain::Instantiate()
          else if (GetOption("StiTestSimpleTube"))       mk->SetAttr("StiTestSimpleTube", true);
          else if (GetOption("StiTestSimpleTubeSector")) mk->SetAttr("StiTestSimpleTubeSector", true);
          else if (GetOption("StiTestEnclosedTubes"))    mk->SetAttr("StiTestEnclosedTubes", true);
+         else if (GetOption("StiTestOverlapTubes"))     mk->SetAttr("StiTestOverlapTubes", true);
+         else if (GetOption("StiTestNoOverlapTubes"))   mk->SetAttr("StiTestNoOverlapTubes", true);
 
          mk->SetAttr("useTestGeom", kTRUE);
          mk->SetAttr("activeTestGeom", kFALSE);
-- 
1.8.2.GIT


From 16ab10745fd7b45d01c7a9a117d3b6960b10ffcb Mon Sep 17 00:00:00 2001
From: Dmitri Smirnov <d.s@plexoos.com>
Date: Mon, 28 Apr 2014 18:04:07 -0400
Subject: [PATCH 5/6] Add BFC option for StiMaker to build ideal geometry

---
 BigFullChain.h | 2 ++
 StBFChain.cxx  | 4 ++++
 2 files changed, 6 insertions(+)

diff --git a/BigFullChain.h b/BigFullChain.h
index 91921fc..491c673 100644
--- a/BigFullChain.h
+++ b/BigFullChain.h
@@ -1472,6 +1472,8 @@ Bfc_st BFC[] = {
   {"StiSst"   ,"","",""                        ,"","StSsdUtil,StSsdDbMaker,StiSsd","Load StiSsd(t)",kFALSE},
   {"StiPxl"   ,"","",""                           ,"","StPxlUtil,StPxlDbMaker,StiPxl","Load StiPxl",kFALSE},
   {"StiIst"   ,"","",""                           ,"","StIstUtil,StIstDbMaker,StiIst","Load StiIst",kFALSE},
+  {"StiIdealGeom", "", "", "", "","",
+                                 "Option to build Sti geometry using default or DB transormations", kFALSE},
   {"StiTestSimpleBox", "", "", "", "",              "", "Flag to build specific test Sti geometry", kFALSE},
   {"StiTestSimpleBox2", "", "", "", "",             "", "Flag to build specific test Sti geometry", kFALSE},
   {"StiTestSimpleBox8", "", "", "", "",             "", "Flag to build specific test Sti geometry", kFALSE},
diff --git a/StBFChain.cxx b/StBFChain.cxx
index fd39171..22c7e2c 100644
--- a/StBFChain.cxx
+++ b/StBFChain.cxx
@@ -464,6 +464,10 @@ Int_t StBFChain::Instantiate()
 	mk->SetAttr("useSst"	 ,kTRUE);
 	mk->SetAttr("activeSst"  ,kTRUE);
       }
+
+      if (GetOption("StiIdealGeom")) mk->SetAttr("buildIdealStiGeom", true);
+      else                           mk->SetAttr("buildIdealStiGeom", false);
+
       if (GetOption("TestIT")){
 
          if      (GetOption("StiTestSimpleBox"))        mk->SetAttr("StiTestSimpleBox", true);
-- 
1.8.2.GIT


From 9d214b1b4aff375863bb89db0c40439f97df3a7a Mon Sep 17 00:00:00 2001
From: Dmitri Smirnov <d.s@plexoos.com>
Date: Tue, 29 Apr 2014 14:30:09 -0400
Subject: [PATCH 6/6] First attempt to build options for using SSD hits

---
 BigFullChain.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/BigFullChain.h b/BigFullChain.h
index 491c673..4082298 100644
--- a/BigFullChain.h
+++ b/BigFullChain.h
@@ -1371,7 +1371,10 @@ Bfc_st BFC[] = {
   {"istHit"     , "" , "" , "istUtil event istDb" , "StIstHitMaker"         , "StIstHitMaker"                           , "Load and run StIstHitMaker"         , kFALSE},
   {"istQA"      , "" , "" , "istUtil StEvent"     , "StIstQAMaker"          , "StIstQAMaker"                            , "Produce QA histograms for IST"      , kFALSE},
   {"istCalib"   , "" , "" , "istUtil"             , "StIstCalibrationMaker" , "StIstCalibrationMaker"                   , "Load and run StIstCalibrationMaker" , kFALSE},
+  {"sstHit"     , "" , "" , "ssddat"              , "StSsdPointMaker"       , "StSsdPointMaker"                         , "Load and run StSsdPointMaker"       , kFALSE},
+
   {"HftMatTree" , "" , "" , "pxlUtil"             , "HftMatchedTree"        , "StHftPoolEventT StHftPoolHftMatchedTree" , "Create HftMatchedTree"              , kFALSE},
+
   {"ssddat"      ,"","","ssd_daq"                             ,"","","SSD full chain for Real Data",kFALSE},
   {"ssd_daq","","","ssdCalDb,svt_T,-sls,-spa,ssdUtil","StSsdDaqMaker","StSsdDaqMaker","... SSD Daq",kFALSE},
   {"ssdfast"     ,"","","ssdDb,StMcEvent,StEvent","StSsdFastSimMaker","StSsdFastSimMaker",
-- 
1.8.2.GIT

