Index: StRoot/StEvent/StContainers.cxx
===================================================================
RCS file: /afs/rhic.bnl.gov/star/packages/repository/StRoot/StEvent/StContainers.cxx,v
retrieving revision 2.30
diff -u -r2.30 StContainers.cxx
--- StRoot/StEvent/StContainers.cxx	23 Jul 2013 11:21:49 -0000	2.30
+++ StRoot/StEvent/StContainers.cxx	28 Jan 2014 01:12:47 -0000
@@ -149,6 +149,7 @@
 #include "StFgtPoint.h"
 #include "StFgtHit.h"
 #include "StPxlHit.h"
+#include "StIstHit.h"
 
 StCollectionImp(Object)
 StCollectionImp(CalibrationVertex)
@@ -206,4 +207,4 @@
 StCollectionImp(FgtStrip)
 StCollectionImp(FgtPoint)
 StCollectionImp(PxlHit)
-
+StCollectionImp(IstHit)
Index: StRoot/StEvent/StContainers.h
===================================================================
RCS file: /afs/rhic.bnl.gov/star/packages/repository/StRoot/StEvent/StContainers.h,v
retrieving revision 2.33
diff -u -r2.33 StContainers.h
--- StRoot/StEvent/StContainers.h	23 Jul 2013 11:21:49 -0000	2.33
+++ StRoot/StEvent/StContainers.h	28 Jan 2014 01:12:47 -0000
@@ -165,6 +165,7 @@
 class StFgtStrip;
 class StFgtHit;
 class StFgtPoint;
+class StIstHit;
 
 StCollectionDef(Object)
 StCollectionDef(BTofHit)
@@ -222,6 +223,6 @@
 StCollectionDef(FgtHit)
 StCollectionDef(FgtStrip)
 StCollectionDef(FgtPoint)
-
+StCollectionDef(IstHit)
 
 #endif
Index: StRoot/StEvent/StEnumerations.h
===================================================================
RCS file: /afs/rhic.bnl.gov/star/packages/repository/StRoot/StEvent/StEnumerations.h,v
retrieving revision 2.55
diff -u -r2.55 StEnumerations.h
--- StRoot/StEvent/StEnumerations.h	23 Jul 2013 11:21:49 -0000	2.55
+++ StRoot/StEvent/StEnumerations.h	13 Mar 2014 22:00:49 -0000
@@ -527,6 +527,17 @@
     kFgtKeepStrip
 };
 
+/** For more IST related constants see StRoot/StIstUtil/StistConsts.h */
+namespace StIstConsts
+{
+   const int kIstNumLadders = 24;          ///< 24 IST Ladders
+   const int kIstNumSensorsPerLadder = 6;  ///< 6 sensor per one IST Ladder
+   const int kIstNumSensors = 144;         ///< 144 sensors
+   const int kIstNumPadsPerSensor = 768;   ///< 768 pads in each sensor
+   const int kIstNumRowsPerSensor = 64;    ///< 64 rows in r-phi direction per each sensor
+   const int kIstNumColumnsPerSensor = 12; ///< 12 columns in beam direction per each sensor
+}
+
 const char *detectorNameById(StDetectorId id);
 StDetectorId detectorIdByName(const char *name);
 #endif
Index: StRoot/StEvent/StEvent.cxx
===================================================================
RCS file: /afs/rhic.bnl.gov/star/packages/repository/StRoot/StEvent/StEvent.cxx,v
retrieving revision 2.53
diff -u -r2.53 StEvent.cxx
--- StRoot/StEvent/StEvent.cxx	5 Mar 2013 14:49:44 -0000	2.53
+++ StRoot/StEvent/StEvent.cxx	28 Jan 2014 01:12:47 -0000
@@ -221,6 +221,7 @@
 #include "StHltEvent.h"
 #include "StFgtCollection.h"
 #include "StPxlHitCollection.h"
+#include "StIstHitCollection.h"
 #include "StTrackNode.h"
 #include "StTrack.h"
 
@@ -779,6 +780,22 @@
     return fgtCollection;
 }
 
+StIstHitCollection*
+StEvent::istHitCollection()
+{
+    StIstHitCollection *istHitCollection = 0;
+    _lookup(istHitCollection, mContent);
+    return istHitCollection;
+}
+
+const StIstHitCollection*
+StEvent::istHitCollection() const
+{
+    StIstHitCollection *istHitCollection = 0;
+    _lookup(istHitCollection, mContent);
+    return istHitCollection;
+}
+
 StPxlHitCollection*
 StEvent::pxlHitCollection()
 {
@@ -1228,6 +1245,12 @@
 }
 
 void
+StEvent::setIstHitCollection(StIstHitCollection* val)
+{
+    _lookupAndSet(val, mContent);
+}
+
+void
 StEvent::setPxlHitCollection(StPxlHitCollection* val)
 {
     _lookupAndSet(val, mContent);
@@ -1341,6 +1364,7 @@
     cout << "\tStFtpcHitCollection:         " << static_cast<void*>(ftpcHitCollection());
     cout << "\tStSvtHitCollection:          " << static_cast<void*>(svtHitCollection());
     cout << "\tStSsdHitCollection:          " << static_cast<void*>(ssdHitCollection());
+    cout << "\tStIstHitCollection:          " << static_cast<void*>(istHitCollection());
     cout << "\tStPxlHitCollection:          " << static_cast<void*>(pxlHitCollection());
     cout << "\tStEmcCollection:             " << static_cast<void*>(emcCollection());
     cout << "\tStFmsCollection:             " << static_cast<void*>(fmsCollection());
@@ -1366,6 +1390,7 @@
     cout << "\t# of FTPC hits:              " << (ftpcHitCollection() ? ftpcHitCollection()->numberOfHits() : 0) << endl;
     cout << "\t# of SVT hits:               " << (svtHitCollection() ? svtHitCollection()->numberOfHits() : 0) << endl;
     cout << "\t# of SSD hits:               " << (ssdHitCollection() ? ssdHitCollection()->numberOfHits() : 0) << endl;
+    cout << "\t# of IST hits:               " << (istHitCollection() ? istHitCollection()->numberOfHits() : 0) << endl;
     cout << "\t# of PXL hits:               " << (pxlHitCollection() ? pxlHitCollection()->numberOfHits() : 0) << endl;
     cout << "\t# of track nodes:            " << trackNodes().size() << endl;
     cout << "\t# of primary tracks:         " << (primaryVertex(0) ? primaryVertex(0)->numberOfDaughters() : 0) << endl;
Index: StRoot/StEvent/StEvent.h
===================================================================
RCS file: /afs/rhic.bnl.gov/star/packages/repository/StRoot/StEvent/StEvent.h,v
retrieving revision 2.46
diff -u -r2.46 StEvent.h
--- StRoot/StEvent/StEvent.h	5 Mar 2013 14:42:45 -0000	2.46
+++ StRoot/StEvent/StEvent.h	28 Jan 2014 01:12:47 -0000
@@ -200,6 +200,7 @@
 class StHltEvent;
 class StFgtCollection;
 class StPxlHitCollection;
+class StIstHitCollection;
 
 class StEvent : public StXRefMain {
 public:
@@ -261,6 +262,8 @@
     const StFgtCollection*              fgtCollection() const;
     StPxlHitCollection*                 pxlHitCollection();
     const StPxlHitCollection*           pxlHitCollection() const;
+    StIstHitCollection*                 istHitCollection();
+    const StIstHitCollection*           istHitCollection() const;
     
     StL0Trigger*                        l0Trigger();
     const StL0Trigger*                  l0Trigger() const;
@@ -331,6 +334,7 @@
     void setSvtHitCollection(StSvtHitCollection*);
     void setSsdHitCollection(StSsdHitCollection*);
     void setPxlHitCollection(StPxlHitCollection*);
+    void setIstHitCollection(StIstHitCollection*);
     void setEmcCollection(StEmcCollection*);
     void setFmsCollection(StFmsCollection*);
     void setRichCollection(StRichCollection*);
Index: StRoot/StEvent/StEventClusteringHints.cxx
===================================================================
RCS file: /afs/rhic.bnl.gov/star/packages/repository/StRoot/StEvent/StEventClusteringHints.cxx,v
retrieving revision 2.33
diff -u -r2.33 StEventClusteringHints.cxx
--- StRoot/StEvent/StEventClusteringHints.cxx	5 Mar 2013 14:42:45 -0000	2.33
+++ StRoot/StEvent/StEventClusteringHints.cxx	28 Jan 2014 01:12:47 -0000
@@ -162,6 +162,7 @@
     setBranch("StRpsCollection",             "evt_aux",      7);
     setBranch("StSsdHitCollection",          "evt_hits",     8);
     setBranch("StSvtHitCollection",          "evt_hits",     8);
+    setBranch("StIstHitCollection",          "evt_hits",     8);
     setBranch("StPxlHitCollection",          "evt_hits",     8);
     setBranch("StTpcHitCollection",          "evt_hits",     8);
     setBranch("StFtpcHitCollection",         "evt_hits",     8);
@@ -185,6 +186,7 @@
     setBranch("StRpsCollection",             "event", 1);
     setBranch("StSsdHitCollection",          "event", 1);
     setBranch("StSvtHitCollection",          "event", 1);
+    setBranch("StIstHitCollection",          "event", 1);
     setBranch("StPxlHitCollection",          "event", 1);
     setBranch("StTpcHitCollection",          "event", 1);
     setBranch("StFtpcHitCollection",         "event", 1);
Index: StRoot/StEvent/StEventLinkDef.h
===================================================================
RCS file: /afs/rhic.bnl.gov/star/packages/repository/StRoot/StEvent/StEventLinkDef.h,v
retrieving revision 2.33
diff -u -r2.33 StEventLinkDef.h
--- StRoot/StEvent/StEventLinkDef.h	5 Mar 2013 14:42:45 -0000	2.33
+++ StRoot/StEvent/StEventLinkDef.h	28 Jan 2014 01:12:47 -0000
@@ -149,6 +149,7 @@
 #pragma link C++ class TrgOfflen2013+;
 
 #pragma link C++ function operator<<(ostream&, const StPxlHit&);        
+#pragma link C++ function operator<<(ostream&, const StIstHit&); 
 #pragma link C++ function operator<<(ostream&, const StTpcPixel&);        
 #pragma link C++ function operator<<(ostream&, const StMeasuredPoint&);        
 #pragma link C++ function operator<<(ostream&, const StHit&);        
Index: StRoot/StEvent/StEventTypes.h
===================================================================
RCS file: /afs/rhic.bnl.gov/star/packages/repository/StRoot/StEvent/StEventTypes.h,v
retrieving revision 2.43
diff -u -r2.43 StEventTypes.h
--- StRoot/StEvent/StEventTypes.h	23 Jul 2013 11:21:49 -0000	2.43
+++ StRoot/StEvent/StEventTypes.h	29 Jan 2014 02:23:48 -0000
@@ -278,5 +278,8 @@
 #include "StFgtHitCollection.h"
 #include "StFgtPoint.h"
 #include "StFgtPointCollection.h"
- 
+#include "StIstHit.h"
+#include "StIstSensorHitCollection.h" 
+#include "StIstLadderHitCollection.h"
+#include "StIstHitCollection.h"
 #endif
