Index: StRoot/StEventUtilities/StEventHitIter.cxx
===================================================================
RCS file: /afs/rhic.bnl.gov/star/packages/repository/StRoot/StEventUtilities/StEventHitIter.cxx,v
retrieving revision 1.4
diff -u -r1.4 StEventHitIter.cxx
--- StRoot/StEventUtilities/StEventHitIter.cxx	8 Jan 2013 21:50:17 -0000	1.4
+++ StRoot/StEventUtilities/StEventHitIter.cxx	28 Jan 2014 01:30:47 -0000
@@ -17,6 +17,7 @@
 #include "StFgtCollection.h"
 #include "StFgtHitCollection.h"
 #include "StFgtPointCollection.h"
+#include "StIstHitCollection.h"
 //________________________________________________________________________________
 StHitIter::StHitIter()
 {
@@ -446,6 +447,82 @@
   return ((StSsdWaferHitCollection*)fCont)->hits().size();
 }
 
+//_______IST_______IST_______IST_______IST_______IST_______IST_______IST_______IST
+//________________________________________________________________________________
+//..............................................................................
+class StIstHitIter : public StHitIter {
+public:
+                StIstHitIter();
+virtual        ~StIstHitIter(){;}
+virtual const TObject *Reset(const TObject *cont);
+virtual const TObject *GetObject (int idx) const;
+virtual           int  GetSize () const;
+  StDetectorId DetectorId() const {return kIstId;}
+protected:
+};
+//..............................................................................
+class StIstLadderHitIter : public StHitIter {
+public:
+                StIstLadderHitIter();
+virtual        ~StIstLadderHitIter(){;}
+virtual const TObject *GetObject (int idx) const;
+virtual           int  GetSize () const;
+protected:
+};
+//..............................................................................
+class StIstSensorHitIter : public StHitIter {
+public:
+                StIstSensorHitIter();
+virtual        ~StIstSensorHitIter(){;}
+virtual const TObject *GetObject (int idx) const;
+virtual           int  GetSize () const;
+protected:
+};
+//________________________________________________________________________________
+StIstHitIter::StIstHitIter()
+{
+  StHitIter *iter = this,*jter=0;
+  iter->SetDowIter((jter=new StIstLadderHitIter())); iter=jter;
+  iter->SetDowIter((jter=new StIstSensorHitIter ()));
+}
+//________________________________________________________________________________
+const TObject *StIstHitIter::Reset(const TObject *cont)
+{
+  const StIstHitCollection *to = 0;
+  if (cont) to = ((StEvent*)cont)->istHitCollection();
+  return StHitIter::Reset(to);
+}
+//________________________________________________________________________________
+const TObject *StIstHitIter::GetObject (int idx) const
+{
+  return (const TObject*)((StIstHitCollection*)fCont)->ladder(idx);
+}
+//________________________________________________________________________________
+int StIstHitIter::GetSize () const
+{
+  return ((StIstHitCollection*)fCont)->numberOfLadders();
+}
+//________________________________________________________________________________
+const TObject *StIstLadderHitIter::GetObject (int idx) const
+{
+  return (const TObject*)((StIstLadderHitCollection*)fCont)->sensor(idx);
+}
+//________________________________________________________________________________
+int StIstLadderHitIter::GetSize () const
+{
+  return ((StIstLadderHitCollection*)fCont)->numberOfSensors();
+}
+//________________________________________________________________________________
+const TObject *StIstSensorHitIter::GetObject (int idx) const
+{
+  return (const TObject*)((StIstSensorHitCollection*)fCont)->hits().at(idx);
+}
+//________________________________________________________________________________
+int StIstSensorHitIter::GetSize () const
+{
+  return ((StIstSensorHitCollection*)fCont)->hits().size();
+}
+
 //________________________________________________________________________________
 //_______FTPC______FTPC______FTPC______FTPC______FTPC______FTPC______FTPC______TPC
 //________________________________________________________________________________
@@ -671,7 +748,7 @@
    case kFtpcEastId:
      Add(new StFtpcHitIter());break;
    case kPxlId: 
-   case kIstId: 
+   case kIstId: Add(new StIstHitIter()) ;break; 
      /* case kFgtId: n.b. This will be removing the RnD version of the FGT */ 
    case kFmsId: 
      Add(new StRnDHitIter(detId));break;
