From e1a6f256efd473b9b41c30e310c4399a3bb9d95e Mon Sep 17 00:00:00 2001
From: Dmitri Smirnov <d.s@plexoos.com>
Date: Thu, 20 Mar 2014 14:22:14 -0400
Subject: [PATCH 1/2] IST realted patches for files in StEventUtilities

---
 StEventUtilities/StEventHitIter.cxx | 79 ++++++++++++++++++++++++++++++++++++-
 1 file changed, 78 insertions(+), 1 deletion(-)

diff --git a/StEventUtilities/StEventHitIter.cxx b/StEventUtilities/StEventHitIter.cxx
index 90f9daa..5c3f088 100644
--- a/StEventUtilities/StEventHitIter.cxx
+++ b/StEventUtilities/StEventHitIter.cxx
@@ -17,6 +17,7 @@
 #include "StFgtCollection.h"
 #include "StFgtHitCollection.h"
 #include "StFgtPointCollection.h"
+#include "StIstHitCollection.h"
 //________________________________________________________________________________
 StHitIter::StHitIter()
 {
@@ -446,6 +447,82 @@ int StSsdWaferHitIter::GetSize () const
   return ((StSsdWaferHitCollection*)fCont)->hits().size();
 }
 
+//_______IST_______IST_______IST_______IST_______IST_______IST_______IST_______IST
+//________________________________________________________________________________
+//..............................................................................
+class StIstHitIter : public StHitIter {
+public:
+                StIstHitIter();
+virtual        ~StIstHitIter(){;}
+virtual const TObject *Reset(const TObject *cont);
+virtual const TObject *GetObject (int idx) const;
+virtual           int  GetSize () const;
+  StDetectorId DetectorId() const {return kIstId;}
+protected:
+};
+//..............................................................................
+class StIstLadderHitIter : public StHitIter {
+public:
+                StIstLadderHitIter();
+virtual        ~StIstLadderHitIter(){;}
+virtual const TObject *GetObject (int idx) const;
+virtual           int  GetSize () const;
+protected:
+};
+//..............................................................................
+class StIstSensorHitIter : public StHitIter {
+public:
+                StIstSensorHitIter();
+virtual        ~StIstSensorHitIter(){;}
+virtual const TObject *GetObject (int idx) const;
+virtual           int  GetSize () const;
+protected:
+};
+//________________________________________________________________________________
+StIstHitIter::StIstHitIter()
+{
+  StHitIter *iter = this,*jter=0;
+  iter->SetDowIter((jter=new StIstLadderHitIter())); iter=jter;
+  iter->SetDowIter((jter=new StIstSensorHitIter ()));
+}
+//________________________________________________________________________________
+const TObject *StIstHitIter::Reset(const TObject *cont)
+{
+  const StIstHitCollection *to = 0;
+  if (cont) to = ((StEvent*)cont)->istHitCollection();
+  return StHitIter::Reset(to);
+}
+//________________________________________________________________________________
+const TObject *StIstHitIter::GetObject (int idx) const
+{
+  return (const TObject*)((StIstHitCollection*)fCont)->ladder(idx);
+}
+//________________________________________________________________________________
+int StIstHitIter::GetSize () const
+{
+  return ((StIstHitCollection*)fCont)->numberOfLadders();
+}
+//________________________________________________________________________________
+const TObject *StIstLadderHitIter::GetObject (int idx) const
+{
+  return (const TObject*)((StIstLadderHitCollection*)fCont)->sensor(idx);
+}
+//________________________________________________________________________________
+int StIstLadderHitIter::GetSize () const
+{
+  return ((StIstLadderHitCollection*)fCont)->numberOfSensors();
+}
+//________________________________________________________________________________
+const TObject *StIstSensorHitIter::GetObject (int idx) const
+{
+  return (const TObject*)((StIstSensorHitCollection*)fCont)->hits().at(idx);
+}
+//________________________________________________________________________________
+int StIstSensorHitIter::GetSize () const
+{
+  return ((StIstSensorHitCollection*)fCont)->hits().size();
+}
+
 //________________________________________________________________________________
 //_______FTPC______FTPC______FTPC______FTPC______FTPC______FTPC______FTPC______TPC
 //________________________________________________________________________________
@@ -671,7 +748,7 @@ int StEventHitIter::AddDetector(StDetectorId detId)
    case kFtpcEastId:
      Add(new StFtpcHitIter());break;
    case kPxlId: 
-   case kIstId: 
+   case kIstId: Add(new StIstHitIter()) ;break; 
      /* case kFgtId: n.b. This will be removing the RnD version of the FGT */ 
    case kFmsId: 
      Add(new StRnDHitIter(detId));break;
-- 
1.8.2.GIT


From aecd480db1136f19d3ba474f427b1ac754a5548f Mon Sep 17 00:00:00 2001
From: Dmitri Smirnov <d.s@plexoos.com>
Date: Thu, 20 Mar 2014 14:57:10 -0400
Subject: [PATCH 2/2] Replaced function calls by constants as that what they
 were returning anyway

---
 StEventUtilities/StEventHitIter.cxx | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/StEventUtilities/StEventHitIter.cxx b/StEventUtilities/StEventHitIter.cxx
index 5c3f088..498f242 100644
--- a/StEventUtilities/StEventHitIter.cxx
+++ b/StEventUtilities/StEventHitIter.cxx
@@ -2,6 +2,7 @@
 #include <stdlib.h>
 #include <assert.h>
 #include "StEvent.h"
+#include "StEvent/StEnumerations.h"
 #include "StHit.h"
 #include "StEventHitIter.h"
 #include "StTpcHitCollection.h"
@@ -500,7 +501,7 @@ const TObject *StIstHitIter::GetObject (int idx) const
 //________________________________________________________________________________
 int StIstHitIter::GetSize () const
 {
-  return ((StIstHitCollection*)fCont)->numberOfLadders();
+  return kIstNumLadders;
 }
 //________________________________________________________________________________
 const TObject *StIstLadderHitIter::GetObject (int idx) const
@@ -510,7 +511,7 @@ const TObject *StIstLadderHitIter::GetObject (int idx) const
 //________________________________________________________________________________
 int StIstLadderHitIter::GetSize () const
 {
-  return ((StIstLadderHitCollection*)fCont)->numberOfSensors();
+  return kIstNumSensorsPerLadder;
 }
 //________________________________________________________________________________
 const TObject *StIstSensorHitIter::GetObject (int idx) const
-- 
1.8.2.GIT

