From cda8397435c5b033d6f5555bf86d75b9d4ea0d6d Mon Sep 17 00:00:00 2001
From: Dmitri Smirnov <d.s@plexoos.com>
Date: Mon, 21 Apr 2014 18:46:57 -0400
Subject: [PATCH 1/6] Process StiMaker attributes to create test volumes

---
 StiMaker/StiMaker.cxx | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/StiMaker/StiMaker.cxx b/StiMaker/StiMaker.cxx
index 30af478..9e0a0d5 100644
--- a/StiMaker/StiMaker.cxx
+++ b/StiMaker/StiMaker.cxx
@@ -451,6 +451,8 @@ More detailed: 				<br>
 
 #include "StiTpc/StiTpcDetectorBuilder.h"
 #include "StiSvt/StiSvtDetectorBuilder.h"
+#include "StiTest/StiTestDetectorBuilder.h"
+#include "StiTest/StiTestDetectorGroup.h"
 #include "StDetectorDbMaker/StiHitErrorCalculator.h"
 // #include "StiRnD/Ist/StiIstDetectorGroup.h"
 // #include "StiRnD/Ist/StiIstDetectorGroup.h"
@@ -651,6 +653,23 @@ Int_t StiMaker::InitDetectors()
       _toolkit->add(group = new StiIstDetectorGroup(IAttr("activeIst"),SAttr("istInputFile")));
       group->setGroupId(kIstId);  
     }
+   if (IAttr("useTestGeom"))
+   {
+      cout << "StiMaker::Init() -I- Adding detector group: Test" << endl;
+
+      StiTestDetectorBuilder::ETestGeomType geomType = StiTestDetectorBuilder::kDefault;
+
+      if      (IAttr("StiTestSimpleBox"))        geomType = StiTestDetectorBuilder::kSimpleBox;
+      else if (IAttr("StiTestSimpleBox2"))       geomType = StiTestDetectorBuilder::kSimpleBox2;
+      else if (IAttr("StiTestSimpleBox8"))       geomType = StiTestDetectorBuilder::kSimpleBox8;
+      else if (IAttr("StiTestSimplePlane"))      geomType = StiTestDetectorBuilder::kSimplePlane;
+      else if (IAttr("StiTestSimpleTube"))       geomType = StiTestDetectorBuilder::kSimpleTube;
+      else if (IAttr("StiTestSimpleTubeSector")) geomType = StiTestDetectorBuilder::kSimpleTubeSector;
+      else if (IAttr("StiTestEnclosedTubes"))    geomType = StiTestDetectorBuilder::kEnclosedTubes;
+
+      _toolkit->add(group = new StiTestDetectorGroup(geomType));
+      group->setGroupId(kTestGeomId);
+   }
 //  if (IAttr("useBTof"))
 //     { 
 //       cout<<"StiMaker::Init() -I- Adding detector group:BTof"<<endl;  
-- 
1.8.2.GIT


From 61ac04d5ff67d1bba7cd0d9e17914e66f476a041 Mon Sep 17 00:00:00 2001
From: Dmitri Smirnov <d.s@plexoos.com>
Date: Wed, 23 Apr 2014 16:12:43 -0400
Subject: [PATCH 2/6] Added new options for Sti test to check overlapping and
 non-overlapping tubes

---
 StiMaker/StiMaker.cxx | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/StiMaker/StiMaker.cxx b/StiMaker/StiMaker.cxx
index 9e0a0d5..b91e170 100644
--- a/StiMaker/StiMaker.cxx
+++ b/StiMaker/StiMaker.cxx
@@ -666,6 +666,8 @@ Int_t StiMaker::InitDetectors()
       else if (IAttr("StiTestSimpleTube"))       geomType = StiTestDetectorBuilder::kSimpleTube;
       else if (IAttr("StiTestSimpleTubeSector")) geomType = StiTestDetectorBuilder::kSimpleTubeSector;
       else if (IAttr("StiTestEnclosedTubes"))    geomType = StiTestDetectorBuilder::kEnclosedTubes;
+      else if (IAttr("StiTestOverlapTubes"))     geomType = StiTestDetectorBuilder::kOverlapTubes;
+      else if (IAttr("StiTestNoOverlapTubes"))   geomType = StiTestDetectorBuilder::kNoOverlapTubes;
 
       _toolkit->add(group = new StiTestDetectorGroup(geomType));
       group->setGroupId(kTestGeomId);
-- 
1.8.2.GIT


From 809226da17674717409ee32eede4dd3845d78fb1 Mon Sep 17 00:00:00 2001
From: Dmitri Smirnov <d.s@plexoos.com>
Date: Mon, 28 Apr 2014 19:26:35 -0400
Subject: [PATCH 3/6] In StiMaker construct the PXL detector using the option
 for ideal/survey geometry transformations. The default is to use a non-ideal
 survey geometry

---
 StiMaker/StiMaker.cxx | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/StiMaker/StiMaker.cxx b/StiMaker/StiMaker.cxx
index b91e170..b0487fa 100644
--- a/StiMaker/StiMaker.cxx
+++ b/StiMaker/StiMaker.cxx
@@ -520,6 +520,8 @@ StiMaker::StiMaker(const Char_t *name) :
   SetAttr("useTracker"          ,kTRUE);
   SetAttr("useVertexFinder"     ,kTRUE);
   SetAttr("Alignment"           ,kFALSE);
+  SetAttr("buildIdealStiGeom"   ,kFALSE);
+
   if (strstr(gSystem->Getenv("STAR"),".DEV"))
      SetAttr("useAux",kTRUE); // Auxiliary info added to output for evaluation
 }
@@ -644,7 +646,7 @@ Int_t StiMaker::InitDetectors()
   if (IAttr("usePixel"))
     {
       cout<<"StiMaker::Init() -I- Adding detector group:PIXEL"<<endl;
-      _toolkit->add(group = new StiPxlDetectorGroup(IAttr("activePixel"),SAttr("pixelInputFile")));
+      _toolkit->add(group = new StiPxlDetectorGroup(IAttr("activePixel"), SAttr("pixelInputFile"), IAttr("buildIdealStiGeom")));
       group->setGroupId(kPxlId);
     }
  if (IAttr("useIst"))
-- 
1.8.2.GIT


From 23168aea005d1a784a64bc29bad81f97ab92e0d8 Mon Sep 17 00:00:00 2001
From: Dmitri Smirnov <d.s@plexoos.com>
Date: Thu, 8 May 2014 15:19:24 -0400
Subject: [PATCH 4/6] my debug

---
 StiMaker/StiDetectorVolume.cxx | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/StiMaker/StiDetectorVolume.cxx b/StiMaker/StiDetectorVolume.cxx
index a4fbb4b..62cf010 100644
--- a/StiMaker/StiDetectorVolume.cxx
+++ b/StiMaker/StiDetectorVolume.cxx
@@ -122,6 +122,9 @@ void StiDetectorVolume::MakeVolume(const StiDetectorBuilder &builder, unsigned i
            LOG_ERROR << "The is no detector for row " << i <<" sector " << j << endm;
            continue;
         }
+
+        LOG_DEBUG << "This detector is: " << next->getName() << " on: "<< next->isOn() << endm;
+
         if (select  &&  (    ( select == kActive   && !next->isActive()) 
                          ||  
                              ( select == kPassivie && next->isActive() )
-- 
1.8.2.GIT


From 34aae56b5be163b9d79b5b47aab18ed43e106d80 Mon Sep 17 00:00:00 2001
From: Dmitri Smirnov <d.s@plexoos.com>
Date: Tue, 20 May 2014 15:36:17 -0400
Subject: [PATCH 5/6] Use SstDbMaker and SstDetectorBuilder when building SST
 in StiMaker

---
 StiMaker/StiMaker.cxx | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/StiMaker/StiMaker.cxx b/StiMaker/StiMaker.cxx
index b0487fa..210e487 100644
--- a/StiMaker/StiMaker.cxx
+++ b/StiMaker/StiMaker.cxx
@@ -435,6 +435,7 @@ More detailed: 				<br>
 #include "StiTpc/StiTpcHitLoader.h"
 #include "StiSvt/StiSvtDetectorGroup.h"
 #include "StiSsd/StiSsdDetectorGroup.h"
+#include "StiSsd/StiSstDetectorGroup.h"
 #include "StiPxl/StiPxlDetectorGroup.h"
 #include "Sti/StiKalmanTrackNode.h"
 #include "Sti/StiKalmanTrack.h"
@@ -470,6 +471,7 @@ More detailed: 				<br>
 #endif
 #include "StTpcDb/StTpcDb.h"
 #include "StSsdDbMaker/StSsdDbMaker.h"
+#include "StSsdDbMaker/StSstDbMaker.h"
 #include "StSvtDbMaker/StSvtDbMaker.h"
 /// Definion of minimal primary vertex errors.
 /// Typical case,vertex got from simulations with zero errors.
@@ -631,9 +633,9 @@ Int_t StiMaker::InitDetectors()
     }
 
   // SSD or SST - they share the db and the kSsdId
-  if (IAttr("useSst") && gStSsdDbMaker){
+  if (IAttr("useSst") && gStSstDbMaker){
     cout<<"StiMaker::Init() -I- Adding detector group:Sst (ssd)"<<endl;
-    _toolkit->add(group = new StiSsdDetectorGroup(IAttr("activeSst"),SAttr("ssdInputFile")));
+    _toolkit->add(group = new StiSstDetectorGroup(IAttr("activeSst"),SAttr("ssdInputFile")));
     group->setGroupId(kSsdId);
 
   } else if ( IAttr("useSsd") && gStSsdDbMaker){
-- 
1.8.2.GIT


From 85b73b09aa6a1063f62117557c339502b70ffa22 Mon Sep 17 00:00:00 2001
From: Dmitri Smirnov <d.s@plexoos.com>
Date: Wed, 21 May 2014 18:12:53 -0400
Subject: [PATCH 6/6] Moved MyClear() to beginning of Make() to allow for
 access of StiTracks by other makers

---
 StiMaker/StiMaker.cxx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/StiMaker/StiMaker.cxx b/StiMaker/StiMaker.cxx
index 210e487..38b845f 100644
--- a/StiMaker/StiMaker.cxx
+++ b/StiMaker/StiMaker.cxx
@@ -747,6 +747,7 @@ Int_t StiMaker::Make()
   try {		// try new event
     
     _tracker->clear();
+    MyClear();
     
     if (mTimg[kHitTimg]) mTimg[kHitTimg]->Start(0);
     _hitLoader->loadEvent(event,_loaderTrackFilter,_loaderHitFilter);
@@ -763,7 +764,6 @@ Int_t StiMaker::Make()
     }
     if (mPullTTree) {iAns = FillPulls();}
     cout<< "StiMaker::Make() -I- Done"<<endl;
-    MyClear();
     if (iAnz) return iAnz;
     if (mTotPrimTks[1] && mTotPrimTks[0]>mTotPrimTks[1]) return kStStop;
     if (IAttr("Alignment") &&  ! _tracker->getNTracks()) return kStErr;
-- 
1.8.2.GIT

